Modules/System/AL Rest Client/src/ALHttpResponseMessage.Codeunit.al

```diff
@@ -0,0 +1,99 @@
+/// <summary>Holder object for the HTTP response data.</summary>
+codeunit 50302 "AL Http Response Message"
+{
+    var
+        HttpResponseMessage: HttpResponseMessage;
+        ALHttpContent: Codeunit "AL Http Content";
+        ErrorMessage: Text;
+        Initialized: Boolean;
+        HasResponse: Boolean;
+        NotInitializedErr: Label 'The Http Response Message has not been initialized';
+
+    internal procedure Initialize()
+    begin
+        Clear(HttpResponseMessage);
+        HasResponse := false;
+        Initialized := true;
+    end;
+
+    /// <summary>Indicates whether the HTTP response message has a success status code.</summary>
+    /// <returns>True if the HTTP response message has a success status code; otherwise, false.</returns>
+    /// <remarks>Any value in the HTTP status code range 2xx is considered to be successful.</remarks>
+    procedure IsSuccessStatusCode() Result: Boolean
```

Comment:
I'm a big fan of using named return values. The reason for that is that I've read somewhere that giving return values a name turns them into assigned variables which theoretically is faster than using unnamed variables. I just can't find the source anymore.
The performance gain is probably very small for a single function but could have an impact when used as much as possible.

Reply:
If you can find that source, I would love to see it, since I was always taught the opposite and if named return values do give a performance gain, I would like to institute that in our organization.

Reply:
I found it: https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-al-complextypes

> It's also possible to use a named return value. Internally, the exit-statement as seen in the example above causes an assignment to an allocated return value. The assignment will have a small performance cost based on the type.

It seems to apply to complex types and not really to simple types. I've just adopted named return variables as a good habit. 🤓